### PR TITLE
Feature/pagination core

### PR DESCRIPTION
**Pull request by VilleMiekkoja**
## Description

1. Implements Pagination (Pager) component for hds-core
2. Fixes theme black supplementary button focus border color (minor visual breaking change)

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1219

## How Has This Been Tested?

- Locally on developers machine

## Screenshots (if appropriate):

image
image
image

👉 [Demo](https://city-of-helsinki.github.io/hds-demo/pagination-core/?path=/story/components-pagination--basic)

### CODE REVIEW

**Review by minevala — COMMENTED** [review]
It works 🎉 
Lots of CSS trickery that I did not know 😁 
I was thinking could we have achieved similar behaviour with flexbox, clamp, etc but I guess we can go with this too

**Comment by ronijaakkola** [comment]
The documentation page was added. Could use some review. I used similar examples to the Storybook but left the customisation example out. I do not know how much that kind of customisation will be done in the future so better not advertise them too much. Also that would be in the "Customisation" tab in the new doc site.

File: packages/core/src/components/pager/pager.css [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  transform: scaleY(-1); /* Trick for flipping the list float direction from bottom to up */
+  --active-page-background-color: var(--color-black);
+  --link-height: 40px;
+}
+
+/* Invisible, floated pseudo-element for reserving space for the hds-pager__button-next,
+we can not use this position for the actual button as it's in incorrect source order for tabbing */
+.hds-pager::before {
+  content: attr(data-next); /* Trick to keep floating element same size as btn--next */
+  content: attr(data-next) / ''; /* Removes this pseudo-element from screen readers where supported */
```
[the remark this answers is not in the document]
Comment (minevala) [reply packages/core/src/components/pager/pager.css]: Ok. This syntax is something I do not quite understand. What is it called?

[the remark this answers is not in the document]
Comment (minevala) [reply packages/core/src/components/pager/pager.css]: Nevermind, found it: 
> /* alt text for generated content, added in the Level 3 specification */
https://developer.mozilla.org/en-US/docs/Web/CSS/content
https://adrianroselli.com/2020/10/alternative-text-for-css-generated-content.html

**Review by minevala — APPROVED** [review]
Nice addition 🏅

File: site/docs/components/pagination.mdx [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+  - the user may need to be able to move to the end of the item list
+- **Use dynamic loading when**
+  - the amount of items in one view is less than 40
+  - divided into pages, you will have less than 3 pages
+  - you have an estimation of how many items there will be
+  - the user does not need to be able to move to the end of the item list
+
+## Accessibility
+- **It is advisable to use colour combinations provided by the implementation.** These combinations are ensured to comply with WCAG AA requirements. When customising colours, refer to [colour guidelines](/design-tokens/colour "Colour") to ensure accessibility.
+- Each page number must an `aria-label` to describe it. For example, `aria-label="Page 3"`.
```
Comment (VilleMiekkoja) [inline site/docs/components/pagination.mdx:58]: must an -> must **have** an

Comment (VilleMiekkoja) [reply site/docs/components/pagination.mdx:58]: - [ ] Add missing have verb

File: site/docs/components/pagination.mdx [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+## Accessibility
+- **It is advisable to use colour combinations provided by the implementation.** These combinations are ensured to comply with WCAG AA requirements. When customising colours, refer to [colour guidelines](/design-tokens/colour "Colour") to ensure accessibility.
+- Each page number must have an `aria-label` to describe it. For example, `aria-label="Page 3"`.
+- The active page number must have an `aria-current="page"` attribute.
+- The truncation element ("..." or ellipsis) does not need to have an `aria-label`.
+
+## Usage and variations
+
+### Basic
+By default, the HDS pagination shows a list of pages and one page is marked as the active page. The previous and next buttons are always shown and they drop under the page number row on smaller screens.
```
Comment (VilleMiekkoja) [inline site/docs/components/pagination.mdx:65]: ... a list of pages and one page ... ->  a list of pages**, and** one page

Comment (VilleMiekkoja) [reply site/docs/components/pagination.mdx:65]: - [ ] Add missing comma

File: packages/core/src/components/pagination/pagination.stories.js [hunk 4]
```diff
@@ -0,0 +1,327 @@
+import './pagination.css';
+
+export default {
+  title: 'Components/Pagination',
+  decorators: [(storyFn) => `<div style="width: 100%;">${storyFn()}</div>`],
+};
+
+export const Basic = () => `
+<div class="hds-pagination-container">
+  <nav class="hds-pagination" role="navigation" aria-label="Pagination" data-next="Next">
```
Comment (minevala) [inline packages/core/src/components/pagination/pagination.stories.js:10]: I'm not sure should we mention in the documentation or in the comments that the data-next attribute is used in styling and it is crucial for the next button to align properly

Comment (VilleMiekkoja) [reply packages/core/src/components/pagination/pagination.stories.js:10]: - [ ] Add note about data-next attribute

**Pull request merged**